### DISCUSSION TITLE
Issue forms dropdown field with options sourced from the list of tags in the repository

### DISCUSSION THREAD

**Discussion by Tyrrrz** [community/community#4299] [Projects and Issues]
Hello!

I'm using GitHub Issue Forms and I have a field called "Version" which the users fill out to indicate which particular version of the software is affected by a bug they are reporting. Right now it's a `type: input` kind of field, but some people end up writing "latest" or similar things instead of putting the actual version there. I want to avoid that.

I want to make it a `type: dropdown` field, but that adds a bit of difficulty when it comes to populating the list of versions as options. All versions on my repo correspond to git tags, so I could write a GitHub Action to do that for me, but I'm not too keen on having to edit the template YAML file and commit it through an action like that. 🙃 

It would be great if you could somehow use the list of tags on a repository as a list of options on an input.

**Comment by lukehefson** [comment]
@Tyrrrz Thanks for the feedback. This is similar (but more specific) to https://github.com/github/feedback/discussions/4266 right?

**Reply by Tyrrrz** [reply]
@lukehefson if I understand that issue correctly, yes. That post doesn't mention where the "structured data" is supposed to be sourced from, so I guess my suggestion is somewhat of an extension to that one 🙂 
I just thought that, given the versions are already reflected in tags, and tags are a natural part of the repository, it would make sense to be able to use that as a data source somehow.

**Reply by lukehefson** [reply]
Totally! All your extra detail and workflow really helps us to think about how we might build and prioritise this. Thanks again @Tyrrrz!!

**Comment by ShaMan123** [comment]
+1
Having a `type: version` makes lot's of sense IMO.
Most, if not all, bug reports need it.
Having a field `registry: npm | github-tags | ...` that fetches the tags and `order: asc | desc` seems enough.
But in general an option to populate data that triggers an update to the element with a js function (running in the browser or even before from nodejs) similar to github node action sounds amazing.

**Reply by ShaMan123** [reply]
https://github.com/ShaMan123/gha-populate-form-version
issue forms version

**Reply by ShaMan123** [reply]
#29067

**Comment by joffrey-bion** [comment]
I would love to have this ability too! There are mostly 2 things I need:

* provide a dropdown with the tags or releases of the current repo (so people can report bugs on specific versions of my own lib)
* provide a dropdown with the versions of another software (some language or framework that is expected to be used in conjunction with my own library)

The idea of a `type: version` + different types of sources for the list of options is great! A few types of version-list providers I could think of:

* `git-tags`: git tags for the current repo (or another git repository)
* `github-releases`: GitHub releases for the current repo (or another GitHub repository)
* `docker-tags`: list of tags for a Docker image repository in DockerHub or a specific registry
* `maven-versions`: list of versions for an artifact published in Maven Central (or another maven repository, given by URL)
* `npm-versions`: list of verisons for an artifact published on npm
* `*-versions`: for other types of packages similarly to the above
